paddle/platform/cuda.h

```diff
@@ -34,6 +34,16 @@ int GetDeviceCount(void) {
   return count;
 }
 
+int GetCurrentDeviceId(void) {
+  int device_id;
+  throw_on_error(cudaGetDevice(&device_id), "cudaGetDevice failed");
```

Comment:
Should we only use `PADDLE_ENFORCE` and not throw `thrust::system_error`?

But it seems that we need to move `enforce` to `platform` if we want to use `PADDLE_ENFORCE` in `platform`